src/client/serialize/daos_serialize.c

```diff
@@ -1175,3 +1245,1833 @@ daos_cont_deserialize_attrs(char *filename, uint64_t *num_attrs,
+	dkey_dset = H5Dopen(hdf5.file, "Dkey Data", H5P_DEFAULT);
+	if (dkey_dset < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed to open dkey data\n");
+		D_GOTO(out, rc);
+	}
+	dkey_dspace = H5Dget_space(dkey_dset);
+	if (dkey_dspace < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed to get dkey dataspace\n");
+		D_GOTO(out, rc);
+	}
+	dkey_vtype = H5Dget_type(dkey_dset);
+	if (dkey_vtype < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed to get dkey vtype\n");
+		D_GOTO(out, rc);
+	}
+	dkey_ndims = H5Sget_simple_extent_dims(dkey_dspace, dkey_dims, NULL);
+	if (dkey_ndims < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed to get dkey dimensions\n");
+		D_GOTO(out, rc);
+	}
+	if (dkey_dims[0] > 0) {
+		D_ALLOC(hdf5.dkey_data, dkey_dims[0] * sizeof(struct dkey));
+		if (hdf5.dkey_data == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+		status = H5Dread(dkey_dset, dkey_vtype, H5S_ALL, H5S_ALL, H5P_DEFAULT,
+				 hdf5.dkey_data);
+		if (status < 0) {
+			rc = -DER_MISC;
+			D_ERROR("Failed to read dkey dataset\n");
+			D_GOTO(out, rc);
+		}
+	}
+
+	/* read akey data */
+	akey_dset = H5Dopen(hdf5.file, "Akey Data", H5P_DEFAULT);
+	if (akey_dset < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed to open akey dataset\n");
+		D_GOTO(out, rc);
+	}
+	akey_dspace = H5Dget_space(akey_dset);
+	if (akey_dspace < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed to get akey dataspace\n");
+		D_GOTO(out, rc);
+	}
+	akey_vtype = H5Dget_type(akey_dset);
+	if (akey_vtype < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed to get akey vtype\n");
+		D_GOTO(out, rc);
+	}
+	akey_ndims = H5Sget_simple_extent_dims(akey_dspace, akey_dims, NULL);
+	if (akey_ndims < 0) {
+		rc = -DER_MISC;
+		D_ERROR("Failed to get akey dimensions\n");
+		D_GOTO(out, rc);
+	}
+	if (akey_dims[0] > 0) {
+		D_ALLOC(hdf5.akey_data, akey_dims[0] * sizeof(struct akey));
+		if (hdf5.akey_data == NULL)
+			D_GOTO(out, rc =  -DER_NOMEM);
+		status = H5Dread(akey_dset, akey_vtype, H5S_ALL, H5S_ALL, H5P_DEFAULT,
+				 hdf5.akey_data);
+		if (status < 0) {
+			rc = -DER_MISC;
+			D_ERROR("Failed to get akey dimensions\n");
+			D_GOTO(out, rc);
+		}
+	}
+
+	/* iterate over read key data from file and write it to a new DAOS container */
+	for (i = 0; i < (int)oid_dims[0]; i++) {
+		oid.lo = hdf5.oid_data[i].oid_low;
+		oid.hi = hdf5.oid_data[i].oid_hi;
+		is_kv = obj_is_kv(oid);
+		if (is_kv) {
+			rc = daos_kv_open(coh, oid, DAOS_OO_RW, &oh, NULL);
+			if (rc != 0) {
+				D_ERROR("failed to open kv object "DF_RC"\n", DP_RC(rc));
+				D_GOTO(out, rc);
+			}
+		} else {
+			rc = daos_obj_open(coh, oid, 0, &oh, NULL);
+			if (rc != 0) {
+				D_ERROR("failed to open object "DF_RC"\n", DP_RC(rc));
+				D_GOTO(out, rc);
+			}
+		}
+
+		dk_off = hdf5.oid_data[i].dkey_offset;
+		dk_next = 0;
+		total_dkeys_this_oid = 0;
+		if (i + 1 < (int)oid_dims[0]) {
+			dk_next = hdf5.oid_data[i + 1].dkey_offset;
+			total_dkeys_this_oid = dk_next - dk_off;
+		} else if (i == ((int)oid_dims[0] - 1)){
+			total_dkeys_this_oid = (int)dkey_dims[0] - (dk_off);
+ 		} 
```

Comment:
(style)  trailing whitespace
(style)  code indent should use tabs where possible
(style)  please, no space before tabs